src/MongoDB.Driver.Core/Core/Servers/TopologyVersion.cs

```diff
@@ -0,0 +1,328 @@
+        /// Comparing topology descriptions freshness does not exhibit the reversal property of
+        /// inequalities e.g. a.IsStalerThanServerResponse(b) (a "&lt;" b) does not imply
+        /// !b.IsStalerThanServerResponse(a) (b "&gt;" a)
+        /// See <seealso cref="CompareFreshnessToServerResponse"/> for more information.
+        /// In the case that this == <paramref name="serverResponse"/>,
+        /// <paramref name="serverResponse"/> will be consider to be fresher.
+        /// </summary>
+        /// <param name="serverResponse">The TopologyVersion received from a server/ </param>
+        /// <returns>
+        /// Wwhether or not this TopologyVersion is fresher than <paramref name="serverResponse"/>.
+        /// </returns>
+        public bool IsFresherThanOrEqualToServerResponse(TopologyVersion serverResponse) => CompareFreshnessToServerResponse(serverResponse) >= 0;
+
+        /// <summary>
+        /// Gets whether or not this TopologyVersion is staler than <paramref name="serverResponse"/>.
+        /// Comparing topology descriptions freshness does not exhibit the reversal property of
+        /// inequalities e.g. a.IsStalerThanServerResponse(b) (a "&lt;" b) does not imply
+        /// !b.IsStalerThanServerResponse(a) (b "&gt;" a).
+        /// See <seealso cref="CompareFreshnessToServerResponse"/> for more information.
+        /// In the case that this == <paramref name="serverResponse"/>,
+        /// <paramref name="serverResponse"/> will be consider to be fresher.
+        /// </summary>
+        /// <param name="serverResponse">The TopologyVersion received from a server/ </param>
+        /// <returns>
+        /// Whether or not this TopologyVersion is staler than <paramref name="serverResponse"/>.
+        /// </returns>
+        public bool IsStalerThanServerResponse(TopologyVersion serverResponse) => CompareFreshnessToServerResponse(serverResponse) <= 0;
+
+        /// <inheritdoc/>
+        public override int GetHashCode() => _hashCode;
+
+        /// <inheritdoc/>
+        public BsonDocument ToBsonDocument() => new BsonDocument {{"processId", _processId}, {"counter", _counter}};
+
+        /// <inheritdoc/>
+        public override string ToString() => $"{{ processId : {_processId}, counter : '{_counter}' }}";
+
+        // explicit interface implementations
+#if NET452 || NETSTANDARD2_0
+        void ISerializable.GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            info.AddValue(nameof(_processId), _processId, typeof(ObjectId));
+            info.AddValue(nameof(_counter), _counter, typeof(long));
+        }
+#endif
+    }
+
+    /// <summary>
+    /// Extensions for Nullable TopologyVersion.
+    /// Provides logic for comparing a TopologyVersion with a null, a null with TopologyVersion, and
+    /// a null TopologyVersion? with a null TopologyVersion?.
+    /// </summary>
+    public static class NullableTopologyVersionExtensions
```

Comment:
Let's get rid of all these extensions methods when `TopologyVersion` is a class instead of a struct.